test/pkg/environment/azure/identity.go

```diff
@@ -0,0 +1,94 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/msi/armmsi"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/samber/lo"
+
+	. "github.com/onsi/gomega"
+)
+
+func (env *Environment) GetTenantID(ctx context.Context) string {
+	cluster, err := env.managedClusterClient.Get(ctx, env.ClusterResourceGroup, env.ClusterName, nil)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(cluster.Identity).ToNot(BeNil())
+	Expect(cluster.Identity.PrincipalID).ToNot(BeNil())
+	return lo.FromPtr(cluster.Identity.TenantID)
+}
+
+func (env *Environment) GetClusterIdentityPrincipalID(ctx context.Context) string {
```

Comment:
minor: Consider `GetClusterIdentity` instead, and then re-use in `GetTenantID` impl? Caller can do `identity.PrincipalID` easily and that gives them access to other things too if they want (like tenant ID).

Reply:
ACK was actually wondering the same thing